src/EGIS.DynamoLib/ShapeFileReader.cs

```diff
@@ -90,12 +113,107 @@ public IEnumerable<Curve> GetAllShapes()
         }
 
         /// <summary>
+        /// Returns all shapes available in this shape file.
+        /// </summary>
+        /// <returns>List of PolyCurve</returns>
+        public IEnumerable<IEnumerable<Curve>> GetAllShapes2()
```

Comment:
We can name this method as GetShapesInAllRecords() just to indicate that the returned shapes are in accordance with the records in the shapefile.